src/mono/wasm/runtime/jiterpreter-support.ts

```diff
@@ -176,14 +177,31 @@ export class WasmBuilder {
             return current.getArrayView(false).slice(0, current.size);
     }
 
+    setImportFunction(name: string, value: Function) {
+        const imp = this.importedFunctions[name];
+        if (!imp)
+            throw new Error("No import named " + name);
+        imp.func = value;
+    }
+
+    getExceptionTag(): any {
+        const exceptionTag = (<any>Module)["asm"]["__cpp_exception"];
+        if (typeof (exceptionTag) !== "undefined")
+            mono_assert(exceptionTag instanceof (<any>WebAssembly).Tag, () => `expected __cpp_exception export from dotnet.wasm to be WebAssembly.Tag but was ${exceptionTag}`);
+        return exceptionTag;
+    }
+
     getWasmImports(): WebAssembly.Imports {
         const memory = (<any>Module).getMemory();
         mono_assert(memory instanceof WebAssembly.Memory, () => `expected heap import to be WebAssembly.Memory but was ${memory}`);
 
+        const exceptionTag = this.getExceptionTag();
         const result: any = {
             c: <any>this.getConstants(),
             m: { h: memory },
         };
+        if (exceptionTag)
+            result.x = { e: exceptionTag };
```

Comment:
I'm not sure `x` is great name if the exception could ever hit or come from customer code. It may conflict there.

Reply:
these 'x' and 'e' names are wasm import names used by jiterpreter modules